### ISSUE TITLE
Waiting for Among Us (arm windows)

### ISSUE THREAD

**Issue by aryairani** [ottomated/CrewLink#605]
# Pre-Flight Checklist
Please use this checklist to avoid spamming:

1. [x] I am not asking a question => use the [Discord](https://discord.gg/9mwuVNA) if you are
2. [x] I am using the Steam version of Among Us
3. [ ] I have tried to use an [alternative voice server](https://status.crewl.ink/)
4. [ ] I have checked everyone in my lobby is on the same CrewLink server
5. [ ] I have used `Ctrl+R` on the CrewLink app when I can't hear some people (this is a known issue)
6. [ ] I have checked that the CrewLink server I'm using is up to date
7. [x] I have a screenshot of any errors
8. [x] I have checked that someone else has not reported this using the [search bar](https://github.com/ottomated/CrewLink/issues?q=is%3Aissue)

**Describe the bug**
I just can't get past "Waiting for Among Us".  I know that's not much to go on.

It doesn't matter whether I start Among Us then CrewLink, CrewLink then Among Us, CrewLink then click "Open Among Us"; it still says "Waiting for Among Us" forever.

**Screenshots**
https://cdn.discordapp.com/attachments/202893438729846793/804895229916348436/unknown.png

**Desktop (please complete the following information):**
 - M1 Macbook Air
 - Parallels Desktop Version 16.3.1 (50393) Technical Preview
 - Windows 10 Pro Insider Preview; 64-bit operating system, ARM-based processor
 - OS Build: 21286.1000

**Additional context**
Here's what I get from the command line:
```
refresh_locale_list [476]: GetKeyboardLayoutList(0, NULL) found 1 layouts.
refresh_locale_list [493]: Received 1 locales.
refresh_locale_list [618]: GetProcAddress() failed for KbdLayerDescriptor!
load_input_helper [866]: refresh_locale_list() found 0 locale(s).
Checking for update
Update for version 2.0.1 is not available (latest version: 2.0.1, downgrade is disallowed).
```

I'm currently installing node.js in the hopes that I can debug the issue myself, but any pointers would be appreciated.

**Comment by aryairani** [comment]
Could it need a new offset store? I don't really know how Microsoft's i386 -> arm translation works.  And is there something I can do to help with that?

**Comment by Stogoh** [comment]
I think it's not related to the offset store. Looks like it cannot fin the proccess. Which version of Among Us do you have (Steam or MS Store)? YOu might also try to launch CrewLink as Administrator.

**Comment by OhMyGuus** [comment]
You should debug the function  checkProcessOpen() and see what getProcesses() returns. 
Other things u should double check if it is running the 32bit version or the 64bit version of crewlink, since someone messaged me before that the installer didn't install it correctly on arm. 
(If u want u can message me on discord so I can help u debug it (ThaGuus#2140))

**Comment by aryairani** [comment]
@Stogoh Ok cool, sorry I had no clue, haha.  I have the Steam version, and also the itch.io version.  I have tried running CrewLink as administrator.  I do have an Among Us.exe process running in the Task Manager.

![image](https://user-images.githubusercontent.com/538571/106402812-6bb59180-63f9-11eb-9dba-4d82efc72824.png)

When I run the `yarn dev` version, it finds and opens Among Us 

![image](https://user-images.githubusercontent.com/538571/106391289-92080c80-63ba-11eb-8ce1-1085d46dc157.png)

but I get this error when I join the lobby:

![image](https://user-images.githubusercontent.com/538571/106391183-17d78800-63ba-11eb-916c-278bca1953fb.png).

Do I need to run my own server to run the dev build?  I haven't changed anything (including version numbers) in package.json; it's still 2.0.1.

When I run the release exe version, I don't get past "Waiting for Among Us".

___
@OhMyGuus I think it is running the 64-bit version of CrewLink and the 32-bit version of Among Us.  

![image](https://user-images.githubusercontent.com/538571/106402846-a1f31100-63f9-11eb-8d76-b9b3dcdf7a1b.png)

When I run the dev version of CrewLink, I think it's the 32-bit version:

![image](https://user-images.githubusercontent.com/538571/106402828-81c35200-63f9-11eb-9405-4a2afded51e8.png)

The dev version gets one step farther for me than the release version does, and but it fails to connect to the voice link server, as shown above.  Thanks for the offer to chat with you on Discord, I will send you a message.

**Comment by aryairani** [comment]
When I run `yarn dist:32` and install from the installer that creates, CrewLink runs no problem.  When I install from the release installer, it doesn't.

**Comment by Stogoh** [comment]
@aryairani

Regarding the isssue with the version missmatch. As you can see in the following screenshot the user agent gets overwritten by "DEV" when you run via `yarn dev`. I think it would be possible to just bypass the code which overwrites the user agent. But I would not recommend to do that on the production server. Therefore I'll suggest that you spin up you own CrewLink server.

![image](https://user-images.githubusercontent.com/2563102/106493277-38d2d280-64b9-11eb-8359-0af4a509cd60.png)

**Comment by aryairani** [comment]
@Stogoh Yes — thank you.  Did you see my message about installing from `yarn dist:32` fixing everything and CrewLink works fine; but installing from the release on github, the install doesn't detect Among Us?

**Comment by Stogoh** [comment]
Intressing. If I run `yarn dev` a 64bit process is created. Try running first `yarn dist:64` then `yarn dev`.

![image](https://user-images.githubusercontent.com/2563102/106497513-6f5f1c00-64be-11eb-9aa5-ba309cc4ab3e.png)

**Comment by aryairani** [comment]
@Stogoh I still get 32-bit electron with `yarn dist:64` then `yarn dev`.  Is there any reason to expect that `yarn dist:64` would affect the runtime environment of `yarn dev` anyway?

**Comment by Stogoh** [comment]
Should not I guess. If I compile with `yarn dist:64` then `yarn dev` CrewLink runs as 64bit. But if I run `yarn dist:32` then `yarn dev` I get some errors regarding 32bit.
